modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +361,106 @@ public void testGetAllIdPs() throws Exception {
+    public void testGetAllPaginatedPaIdPs() throws Exception {
+
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+        Assert.assertNotNull(idpList);
+        log.info("All idp paginated list size : " + idpList.size());
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (idp.getIdentityProviderName().equals(testIdpName) || idp.getIdentityProviderName()
+                        .equals(testIdpNameSearch)) {
+                    filteredIdp = idp;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Added new idps not found in the idp list");
+        } else {
+            Assert.fail("Unable to find added identity provider");
+        }
+    }
+
+    @Test(priority = 6, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation")
+    public void testGetPaginatedPaIdPs() throws Exception {
+
+        String filter = "name sw \"Search\" and name ew \"Test\"";
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        Assert.assertNotNull(idpList);
+        log.info("All filtered paginated idp list size : " + idpList.size());
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            IdentityProvider notFilteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (idp.getIdentityProviderName().equals(testIdpNameSearch)) {
+                    filteredIdp = idp;
+                } else if (idp.getIdentityProviderName().equals(testIdpName)) {
+                    notFilteredIdp = idp;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Searched Idp not found");
+            Assert.assertNull(notFilteredIdp, "Searched Incorrect Idp");
+        } else {
+            Assert.fail("Unable to find filtered identity provider");
+        }
+    }
+
+    @Test(priority = 7, groups = "wso2.is", description = "test getAllIdpCount operation")
+    public void testGetAllIdPsCount() throws Exception {
+
+        Integer idpCount = idpMgtServiceClient.getAllIdpCount();
+        log.info("All idp paginated filtered count : " + idpCount);
+        Assert.assertNotNull(idpCount, "Added new test idp not found in the idp list");
+    }
+
+    @Test(priority = 8, groups = "wso2.is", description = "test getFilteredIdpCount operation")
+    public void testGetAllFilteredIdPsCount() throws Exception {
+
+        String filter = "name sw \"Search\" and name ew \"Test\"";
+        Integer idpCount = idpMgtServiceClient.getFilteredIdpCount(filter);
+        log.info("Idp paginated filtered count : " + idpCount);
+        Assert.assertNotNull(idpCount, "Added new test idp not found in the idp list");
```

Comment:
can;t we assert the idp count with an expected int value in addition to asserting not null?